code/game/gamemodes/meteor/meteors.dm

```diff
@@ -225,12 +231,23 @@ var/global/list/meteors_cataclysm = list(\
 		O.throw_at(dest, 5, 10)
 
 /obj/effect/meteor/proc/meteor_effect()
-	for(var/mob/M in GLOB.player_list)
-		var/turf/T = get_turf(M)
-		if(!T || T.z != src.z)
-			continue
-		var/dist = get_dist(M.loc, src.loc)
-		shake_camera(M, dist > 20 ? 3 : 5, dist > 20 ? 1 : 3)
+	if(flicker_range)
+		for(var/obj/machinery/light/L in range(flicker_range, src))
+			L.flicker(rand(5,15))
+
+	if(shake_range)
+		for(var/mob/M in range(shake_range, src))
+			if(!M.stat && M.client && !istype(M, /mob/living/silicon/ai))\
+				shake_camera(M, 3, 1)
+
+	else
+		for(var/mob/M in GLOB.player_list) //this is an insane way of doing this bay
+			var/turf/T = get_turf(M)
+			if(!T || T.z != src.z)
```

Comment:
For some reason Bay doesn't use that operator much, even though Lummox has said it's more performant. Since I'm moving this code around I might as well change it though